docs/src/concepts/memory-trait.md

```diff
@@ -23,6 +26,13 @@ The `Memory` trait intentionally models a [WebAssembly memory instance](https://
 This design choice ensures consistency with the interface of memories available to canisters.
 It also provides future compatibility with potential multi-memory support in canisters.
 
+## Safety contract  
```

Comment:
Maybe `Panics` is a better section name? Just because it is still memory safe to call these functions with out of bounds access.

Reply:
done.